test/e2e/framework/framework.go

```diff
@@ -396,13 +386,40 @@ func newSingleIngress(name, path, host, ns, service string, port int, annotation
 	}
 
 	if withTLS {
-		ing.Spec.TLS = []extensions.IngressTLS{
+		spec.TLS = []extensions.IngressTLS{
 			{
 				Hosts:      []string{host},
 				SecretName: host,
 			},
 		}
 	}
 
+	return newSingleIngress(name, ns, annotations, spec)
+}
+
+func NewSingleIngressWithBackend(name, ns, service string, port int, annotations *map[string]string) *extensions.Ingress {
```

Comment:
IngressWithBackend is a bit confusing for what this function does. IMHO `NewSingleCatchAllIngress` fits better.